sched/clustermembership/dynamicCluster.go

```diff
@@ -1,49 +1,77 @@
 package cluster_membership
 
-/*
- * Represents a Cluster of nodes that can change over time.
- * Keeps a Map of the nodeId to Node for fast lookup of nodes.
- */
+//
+// Represents a Cluster of nodes that can change over time.
+// Keeps a Map of the nodeId to Node for fast lookup of nodes.
+//
 type dynamicCluster struct {
 	memberSet   map[string]bool
 	membersList []Node
+	updateCh    chan NodeUpdate
```

Comment:
No, I think you're right to have one channel handle all updates.

If you have multiple watchers, then each update would only be seen by one watcher.

Also, there might be a race condition where you ask for the members and then start listening, but miss updates? Maybe there should be one method that, atomically, gives you the current state and  a new channel?